### PR TITLE
E1608. Refactor & write unit tests for [question_type].rb (eg. criterion.rb)

### PR DESCRIPTION
**Pull request by hawkwoodye**
[Google Doc](https://docs.google.com/document/d/1s-pGuijZ_vvhQlc8HjgOmTYDe_UBwF56UP7q94slhEI/edit?ts=56eddfeb#)

What needs to be done:
Refactor
1. Criterion.rb
    1.1 Method has too many lines: Line 35 - 158 [106/30]
    1.2 Cyclomatic complexity for complete is too high: Line 35 - 158 [28/6]
    1.3 Identical code found in 1 other location: Line 20 - 32 (mass = 89)
    1.4 Identical code found in 1 other location: Line 135 - 139 (mass = 50)
    1.5 Identical code found in 1 other location Line 13 - 14 (mass = 40)
    1.6  Similar code found in 1 other location Line 103 - 105 (mass = 26)
    1.7 Similar code found in 2 other locations Line 152 - 153 (mass = 22)
    1.8 Similar code found in 3 other locations: Line 130 - 134 (mass = 18)
2. Scale.rb
    2.1 Cyclomatic complexity for complete is too high. Method has too many lines
    2.2 Identical code found in 1 other location: line 17-28
    2.3 Identical code  found in 1 other location: line 50-54
    2.4 Identical code found in 1 other location: line 10-11
    2.5 Similar code found in 3 other locations: line 60-64
3. Checkbox.rb
    3.1 Method has too many lines.
    3.2 Cyclomatic complexity for complete is too high.
    3.3 Similar codes in edit method are found in 2 other locations.
    3.4 Similar codes in view_question_text method are found in 4 other locations.
4. Questionnaire_header.rb
5. Upload_file.rb
Unit-test
1. Criterion.rb
2. Scale.rb
3. Checkbox.rb
4. Upload_file.rb

### CODE REVIEW

File: app/views/response/show_calibration_results_for_student.html.erb [hunk 1]
```diff
@@ -1,101 +0,0 @@
-<h1>Your calibration results <%= " for" + @assignment.course.name if !@assignment.course.nil?%></h1>
```
Comment (yangsong8) [inline app/views/response/show_calibration_results_for_student.html.erb]: Any reason that this was removed?

**Comment by yangsong8** [comment]
Just to confirm, this pull request is for "E1608.  Refactor & write unit tests for [question_type].rb (eg. criterion.rb)"? I saw many places of the code change which are beyond the scope of you project (e.g. changing calibration feature). 

Was this something you were told to your contact person?

File: app/views/response/show_calibration_results_for_student.html.erb (hunk 1 above)
Comment (hawkwoodye) [inline app/views/response/show_calibration_results_for_student.html.erb]: Hi Yang, I think there must be some mistakes of the beginning version we cloned. I think for our project 1608, we only needs to touch question relative rb files. I don't know why the line is removed...

**Comment by hawkwoodye** [comment]
Dear Yang,

As you said, for our project, we only need to touch Question related files,
my guess is that the version/branch we start wrongly included other
people's work... Could you please give me some help of finding the correct
version we should start?

Thanks,
Ye

**Comment by yangsong8** [comment]
Please backup your changed files, fork the current master branch and make the changes again. It does not worth the time to clean your repo. Thanks.

**Comment by Winbobob** [comment]
Why do you delete `placeholder=\"Edit question content here\"` in all [question_type].rb file?

**Comment by hawkwoodye** [comment]
I don't remember I did that... Sorry, it's weird, do you want me to put
them back?

Thanks,
Ye

**Comment by Winbobob** [comment]
I have already merged part of your code, so you do not need to put it back now.

**Comment by hawkwoodye** [comment]
Thanks a lot, Zhewei. Owe you one.
Ye

File: spec/features/peer_review_spec.rb [hunk 2]
```diff
@@ -57,9 +57,9 @@ def load_questionnaire
 
     # Fill in a textbox and a dropdown
     fill_in "responses[0][comment]", :with => "HelloWorld"
-    select 5, :from => "responses[0][score]"
-    click_button "Submit Review"
-    expect(page).to have_content "Your response was successfully saved."
+    #select 5, :from => "responses[0][score]"
```
Comment (yangsong8) [inline spec/features/peer_review_spec.rb]: Why those lines are commented out?

**Pull request closed without merging**